cli/azd/pkg/environment/config.go

```diff
@@ -0,0 +1,66 @@
+package environment
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+// Config contains environment-related configuration.
+type Config = config.Config
+
+type ConfigManager struct {
+	// Path to configuration file on filesystem
+	path    string
+	persist config.Manager
+}
+
+func NewConfigManager(
+	azdCtx azdcontext.AzdContext,
+	envName string,
+	configMgr config.Manager) *ConfigManager {
+	envRoot := azdCtx.EnvironmentRoot(envName)
+	return &ConfigManager{
+		path:    filepath.Join(envRoot, azdcontext.ConfigFileName),
+		persist: configMgr,
+	}
+}
+
+func NewConfigManagerFromRoot(
+	root string,
```

Comment:
Consider renaming func & param from `root` to `path`.  All we know is that we expect a config manager for the specified path and the component itself doesn't need to distinguish what root means.

Another option could be `rootPath`.